### ISSUE TITLE
Coverity build only fails with Travis-CI

### ISSUE THREAD

**Issue by anonimal** [travis-ci/travis-ci#6274]
As reported in https://github.com/monero-project/kovri/issues/209.

tl;dr: the [Travis-CI build](https://travis-ci.org/monero-project/kovri/builds/138629182) is successful but the Coverity build *with* Travis-CI fails. When uploading a tarball directly to Coverity, the build is successful.

Here is Coverity's error when using Travis-CI
> Last Build Status: Failed. Your build has failed due to the following reason. Please fix the error and upload the build again.
Error details: Build uploaded has not been compiled fully. Please fix any compilation error. You may have to run bin/cov-configure as described in the article on Coverity Community. Last few lines of cov-int/build-log.txt should indicate 85% or more compilation units ready for analysis...

I repeat: the Coverity build is successful when uploading the exact same code manually.

In an attempt to debug, because we are using custom exported compiler version variables in Trusty, I would like to try the following suggestion from Coverity:

```cov-configure --comptype gcc --compiler /usr/bin/g++-5```

but I do not see anywhere in the Travis-CI documentation where this would go. I would assume to prepend ```/tmp/cov-configure ...``` in ```build_command_prepend``` before any build commands, but any confirmation of this would be very helpful and greatly appreciated (we use a fork-an-pull model so trial-and-error is cumbersome).

Here is our [yaml](https://github.com/monero-project/kovri/blob/coverity_scan/.travis.yml).

**Comment by rakhimov** [comment]
This seems to be resolved. I submitted my build via Travis: https://scan.coverity.com/projects/2555

**Comment by anonimal** [comment]
> we are using custom exported compiler version variables in Trusty

@rakhimov but you seem to be using default gcc 4.8 - so Coverity gets what it expects. In your linux container, have you tried building with gcc5?

**Comment by rakhimov** [comment]
I see. I don't build with gcc5.
However, I did have the same problem with gcc-4.8 around 2 weeks ago.

**Comment by anonimal** [comment]
No luck. The problem remains.

**Comment by rakhimov** [comment]
Your scan says success 
https://scan.coverity.com/projects/7621/
https://travis-ci.org/monero-project/kovri
Is it only gcc 4.8?

**Comment by anonimal** [comment]
@rakhimov after the Coverity build failed, I immediately performed a manual upload. What you're seeing is the manual upload success. As for the [Travis-CI](https://travis-ci.org/monero-project/kovri/builds/144656306) build success, this is from my original post:

>tl;dr: the Travis-CI build is successful but the Coverity build with Travis-CI fails.

**Comment by BanzaiMan** [comment]
https://travis-ci.org/monero-project/kovri/builds/144656306#L997 indicates that this is the same issue reported as #6316. I think this was/is an issue with Coverity's archive.

Closing this now.

**Comment by anonimal** [comment]
@BanzaiMan thanks for resolving. Can you advise on the tail-end of the OP: is it possible to execute ```cov-configure --comptype gcc --compiler /usr/bin/g++-5``` within Travis and if so, where?

**Comment by BanzaiMan** [comment]
@anonimal Where is the command `cov-configure` supposed to come from? I don't think our build image has it, and we do not install it on the fly for Coverity Scan addon (https://github.com/travis-ci/travis-build/blob/99e149b33de4ff4dc7d8bf694fc8e69b570edf97/lib/travis/build/addons/coverity_scan.rb), so I don't think that is an option.

**Comment by BanzaiMan** [comment]
Correction. If you can download it, and if it can be run any time before the [travisci_build_coverity_scan.sh](https://scan.coverity.com/scripts/travisci_build_coverity_scan.sh) script runs, you can do that in `before_script` or another customizable build phase.

**Comment by anonimal** [comment]
@BanzaiMan ah, perfect, I see now. Thank you for clarifying and linking to the script.
